### PR TITLE
correct graph constructing from edge list

### PR DESCRIPTION
**Pull request by orazve**
# Description
Please include a summary of the change. For large or complex changes please include enough information to introduce your change and explain motivation for it.

Changes proposed in this pull request:
-
-
-

### CODE REVIEW

File: cpp/oneapi/dal/data/undirected_adjacency_array_graph.cpp [hunk 1]
```diff
@@ -122,44 +124,106 @@ void convert_to_csr_impl(const edge_list<vertex_type<G>> &edges, G &g) {
...
+        new daal::services::Atomic<int_t>[layout_unfilt->_vertex_count + 1];
+    int_t total_sum_degrees = 0;
+    rows_cv[0].set(total_sum_degrees);
 
-    for (std::int64_t i = 0; i < layout->_vertex_count; ++i) {
+    for (size_t i = 0; i < layout_unfilt->_vertex_count; ++i) {
+        total_sum_degrees += degrees_cv[i].get();
+        rows_cv[i + 1].set(total_sum_degrees);
+    }
+    delete[] degrees_cv;
```
Comment (bysheva) [inline cpp/oneapi/dal/data/undirected_adjacency_array_graph.cpp:166]: Check for nullptr or remove allocation\deallocation at all

File: cpp/oneapi/dal/data/undirected_adjacency_array_graph.cpp [hunk 2]
```diff
@@ -122,44 +124,106 @@ void convert_to_csr_impl(const edge_list<vertex_type<G>> &edges, G &g) {
...
+                       layout_unfilt->_vertex_count + 1,
+                       [&](int n) {
+                           _rows_un[n] = rows_cv[n].get();
+                       });
+
+    daal::threader_for(edges.size(), edges.size(), [&](int u) {
+        _cols_un[rows_cv[edges[u].first].inc() - 1]  = edges[u].second;
+        _cols_un[rows_cv[edges[u].second].inc() - 1] = edges[u].first;
+    });
+    delete[] rows_cv;
```
Comment (bysheva) [inline cpp/oneapi/dal/data/undirected_adjacency_array_graph.cpp:183]: The same for allocations\deallocations

File: cpp/oneapi/dal/util/load_graph.hpp [hunk 3]
```diff
@@ -19,6 +19,9 @@
 #include "oneapi/dal/data/graph.hpp"
 #include "oneapi/dal/util/csv_data_source.hpp"
 
+#include <chrono>
+#include <iostream>
+
 namespace oneapi::dal::preview {
 
 struct edge_list_to_csr_descriptor {};
```
Comment (bysheva) [inline cpp/oneapi/dal/util/load_graph.hpp:27]: Please align the API for loader and descriptor with the approved one: https://gitlab.devtools.intel.com/tbysheva/graph_tasklist/-/blob/master/Tasklist.md#load-graph

File: cpp/oneapi/dal/util/load_graph.hpp [hunk 4]
```diff
@@ -31,7 +34,23 @@ edge_list<IndexT> load(const std::string &name);
 template <typename G, typename Descriptor, typename DataSource>
 G load_graph(const Descriptor &d, const DataSource &ds) {
     G graph_data;
-    convert_to_csr(detail::load<typename G::vertex_type>(ds.get_filename()), graph_data);
+
+    std::chrono::steady_clock::time_point begin_read = std::chrono::steady_clock::now();
```
Comment (bysheva) [inline cpp/oneapi/dal/util/load_graph.hpp:38]: Need to remove measurements for the final commit

File: cpp/oneapi/dal/data/undirected_adjacency_array_graph.cpp [hunk 5]
```diff
@@ -122,44 +124,106 @@ void convert_to_csr_impl(const edge_list<vertex_type<G>> &edges, G &g) {
...
 
-    layout->_vertex_count = max_id + 1;
-    layout->_degrees      = array<vertex_t>::zeros(layout->_vertex_count);
+    layout_unfilt->_vertex_count = max_node_id + 1;
 
-    for (auto edge : edges) {
-        layout->_degrees[edge.first]++;
-        layout->_degrees[edge.second]++;
-    }
+    daal::services::Atomic<int_t> *degrees_cv =
```
Comment (bysheva) [inline cpp/oneapi/dal/data/undirected_adjacency_array_graph.cpp:145]: Can we use any container for atomics to support the allocator?

File: cpp/oneapi/dal/util/load_graph.hpp (hunk 3 above)
Comment (orazve) [reply cpp/oneapi/dal/util/load_graph.hpp:27]: sorry, wrong branch is pull requested. See https://github.com/bysheva/oneDAL/pull/7

**Comment by orazve** [comment]
wrong branch was pull requested

**Pull request closed without merging**